Godot 3.2.4 RC5 fails to build on FreeBSD

**Godot version:**
1d0929a


**OS/device including version:**
FreeBSD 13-RC2 / AMD64


**Issue description:**

Build process seems to hang multiple times on pthreads.

`
[Initial build] Compiling ==> drivers/gles3/rasterizer_scene_gles3.cpp
drivers/unix/thread_posix.cpp: In function 'Error set_name(const String&)':
drivers/unix/thread_posix.cpp:53:2: error: 'pthread_set_name_np' was not declared in this scope; did you mean 'pthread_setname_np'?
   53 |  pthread_set_name_np(running_thread, p_name.utf8().get_data());
      |  ^~~~~~~~~~~~~~~~~~~
      |  pthread_setname_np
scons: *** [drivers/unix/thread_posix.x11.tools.64.o] Error 1
`


**Steps to reproduce:*
Clone latest Godot 3.2.4 from git. Build with scons -j8 builtin-libvpx=False pulseaudio=False platform=x11

also will fail when adding use_llvm=yes to the command line.

Sorry, was fumbling around and accidentally fatfingered the close issue button. AFAIK now it still doesn't work but I pulled the source from 3.2 master branch, so maybe I didn't get your parch?  I'm not a regular user of github.

This is what I get from building as of pulling the main 3.2 branch about 10 minutes ago:

`[Initial build] Linking Static Library ==> modules/libmodules.x11.tools.64.a
drivers/unix/thread_posix.cpp: In function 'Error set_name(const String&)':
drivers/unix/thread_posix.cpp:53:2: error: 'pthread_set_name_np' was not declared in this scope; did you mean 'pthread_setname_np'?
   53 |  pthread_set_name_np(running_thread, p_name.utf8().get_data());
      |  ^~~~~~~~~~~~~~~~~~~
      |  pthread_setname_np
[Initial build] Compiling ==> drivers/alsa/asound-so_wrap.c
[Initial build] Compiling ==> drivers/alsa/audio_driver_alsa.cpp
[Initial build] Compiling ==> drivers/coreaudio/audio_driver_coreaudio.cpp
scons: *** [drivers/unix/thread_posix.x11.tools.64.o] Error 1
`

It's indeed possible that BSD updated the pthread library was changed, but to make sure this was a regression I built 3.2.3 stable from the ports tree last night. So, even though I'm using a RC level BSD Kernel / Libraries that one still builds, as long as you tell it not to use built-in libvpx.

--- a/drivers/unix/thread_posix.cpp
+++ b/drivers/unix/thread_posix.cpp
@@ -35,6 +35,10 @@
 #include "core/os/thread.h"
 #include "core/string/ustring.h"
 
+#ifdef PTHREAD_BSD_SET_NAME
+#include <pthread_np.h>
+#endif
+
 static Error set_name(const String &p_name) {
 #ifdef PTHREAD_NO_RENAME
 	return ERR_UNAVAILABLE;

